### ISSUE TITLE
No error thrown when uploading an invalid extension

### ISSUE THREAD

**Issue by MordantWastrel** [MasaCMS/MasaCMS#99]
Environment: Docker container running Commandbox & Lucee 5.3.9.141

Converted from Mura 7.1 to Masa 7.3.5. Everything appears to be working, but the file manager is not saving any uploaded files. We are mounting an external file system into the container for our assets folder; this works fine (Masa can read them) but when we try to upload a file, the request returns an object indicating an entry in `failed` but without any explanation as to why. It also has `success : 1` which seems like it should not be the case.

```
{
	"data": {
		"failed": [{
			"filewasrenamed": false,
			"attemptedserverfile": "A050D0B2-D207-11EC-91A3-16CFA4443AF3.jpg",
			"serverfilename": "A050D0B2-D207-11EC-91A3-16CFA4443AF3",
			"timelastmodified": "June, 22 2022 11:31:51 -0700",
			"fileexisted": false,
			"oldfilesize": "119543",
			"serverfileext": "jpg",
			"clientfilename": "A050D0B2-D207-11EC-91A3-16CFA4443AF3",
			"filewasappended": false,
			"timecreated": "June, 22 2022 11:31:51 -0700",
			"filesize": 119543,
			"filewassaved": true,
			"clientfileext": "jpg",
			"contenttype": "image",
			"datelastaccessed": "June, 22 2022 11:31:51 -0700",
			"clientfile": "A050D0B2-D207-11EC-91A3-16CFA4443AF3.jpg",
			"clientdirectory": "",
			"contentsubtype": "jpeg",
			"filewasoverwritten": false,
			"serverdirectory": "/usr/local/lib/serverHome/WEB-INF/lucee-web/temp",
			"serverfile": "A050D0B2-D207-11EC-91A3-16CFA4443AF3.jpg"
		}],
		"success": "1",
		"allowedextensions": "doc,docx,gif,jpeg,pdf,png,ppt,pptx,ppsx,xls,xlsx,xml",
		"uploaded": [{
			"filewasrenamed": false,
			"attemptedserverfile": "A050D0B2-D207-11EC-91A3-16CFA4443AF3.jpg",
			"serverfilename": "A050D0B2-D207-11EC-91A3-16CFA4443AF3",
			"timelastmodified": "June, 22 2022 11:31:51 -0700",
			"fileexisted": false,
			"oldfilesize": "119543",
			"serverfileext": "jpg",
			"clientfilename": "A050D0B2-D207-11EC-91A3-16CFA4443AF3",
			"filewasappended": false,
			"timecreated": "June, 22 2022 11:31:51 -0700",
			"filesize": 119543,
			"filewassaved": true,
			"clientfileext": "jpg",
			"contenttype": "image",
			"datelastaccessed": "June, 22 2022 11:31:51 -0700",
			"clientfile": "A050D0B2-D207-11EC-91A3-16CFA4443AF3.jpg",
			"clientdirectory": "",
			"contentsubtype": "jpeg",
			"filewasoverwritten": false,
			"serverdirectory": "/usr/local/lib/serverHome/WEB-INF/lucee-web/temp",
			"serverfile": "A050D0B2-D207-11EC-91A3-16CFA4443AF3.jpg"
		}],
		"saved": []
	},
	"method": "upload",
	"apiversion": "v1"
}
```

**Comment by grantshepert** [comment]
I would suggest scanning your Lucee "temp" directory to see if the uploaded file is there after the attempt. The file did upload (it has all the server-side information in the dump) so I suspect the issue occurs when moving the file from the "temp" directory to the intended destination. If this is the case, the error should also show up in the Lucee logs.

**Comment by MordantWastrel** [comment]
The only thing in the `temp` directory is the `compress` directory and there are no errors in the Lucee logs; I checked the Commandbox server.out as well as WEB-INF/logs for `lucee-web` and `lucee-server`.

**Comment by grantshepert** [comment]
Thanks for the update! 

Keep in mind, what you are seeing in the response is in context of a process that supports multiple file upload.

So, "Success" is a bit abstract, as the process of attempting to upload the image(s) can be successful, even if one or more images aren't accepted (i.e. the flag is for indicating programatic/functional, not process, errors). Some of the other flags (i.e. "filewassaved") can also be misleading because files are first saved in a temp directory, then checked for allowed/etc ... so, good for debugging but misleading otherwise.

All that said, adding an error message that lists the image(s) that fail during upload is obv. a good thing.

**Comment by MordantWastrel** [comment]
>So, "Success" is a bit abstract, as the process of attempting to upload the image(s) can be successful,

As a technical matter, I appreciate what you're trying to say, that -- in the process of some task X that has 3 steps, you may succeed in the first 2 steps and therefore 'have some success' before it fails on step 3. As a user, that's not relevant.

I want to upload one or more files.

Either my uploads succeed or fail.

If the 'upload' part succeeds but the 'put the file somewhere Masa can actually use it' part fails, the whole thing has failed. Masa should a) communicate that it failed and b) tell me why it failed.

**Comment by guustnieuwenhuis** [comment]
We've fixed this issue in release 7.3.8
